mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -230,4 +264,87 @@ private int getClusteringColumnIndex(
       return -1;
     }
   }
+
+  private CqlNameGenerator buildCqlNameGenerator(TypeElement classElement) {
+
+    NamingStrategy namingStrategy = classElement.getAnnotation(NamingStrategy.class);
+    if (namingStrategy == null) {
+      return CqlNameGenerator.DEFAULT;
+    }
+
+    NamingConvention[] conventions = namingStrategy.convention();
+    TypeMirror[] customConverterClasses = readCustomConverterClasses(classElement);
+
+    if (conventions.length > 0 && customConverterClasses.length > 0) {
+      context
+          .getMessager()
+          .error(
+              classElement,
+              "Invalid annotation configuration: %s must have either a 'convention' "
+                  + "or 'customConverterClass' argument, but not both",
+              NamingStrategy.class.getSimpleName());
+      // Return a generator anyway, so that the processor doesn't crash downstream
+      return new CqlNameGenerator(conventions[0]);
+    } else if (conventions.length == 0 && customConverterClasses.length == 0) {
+      context
+          .getMessager()
+          .error(
+              classElement,
+              "Invalid annotation configuration: %s must have either a 'convention' "
+                  + "or 'customConverterClass' argument",
+              NamingStrategy.class.getSimpleName());
+      return CqlNameGenerator.DEFAULT;
+    } else if (conventions.length > 0) {
+      if (conventions.length > 1) {
+        context
+            .getMessager()
+            .warn(
+                classElement,
+                "Too many naming conventions: %s must have at most one 'convention' "
+                    + "argument (will use the first one: %s)",
+                NamingStrategy.class.getSimpleName(),
+                conventions[0]);
+      }
+      return new CqlNameGenerator(conventions[0]);
+    } else {
+      if (customConverterClasses.length > 1) {
+        context
+            .getMessager()
+            .warn(
+                classElement,
+                "Too many custom converters: %s must have at most one "
+                    + "'customConverterClass' argument (will use the first one: %s)",
```

Comment:
Specifying multiple classes is already a user error, hopefully the user will notice the warning and fix the issue.

Arguably we could issue an error instead of a warning for this kind of mistake.

Reply:
But to answer your original question, yes, [the order is deterministic](https://docs.oracle.com/javase/7/docs/api/javax/lang/model/element/AnnotationValue.html):
> representing the elements, in declared order, if the value is an array